app/controllers/application_controller.rb

```diff
@@ -12,4 +12,12 @@ def after_sign_in_path_for(_resource)
   def error_render_method
     render(template: 'errors/timeout', status: 408) && return
   end
+
+  def authenticate_wimi
+    redirect_to root_path, alert: 'You don\'t have the necessary rights to perform this operation' unless current_user && (current_user.admin? || current_user.wimi?)
```

Comment:
is the root_path here always the best option? can it happen that a user is on a specific site and will be redirected to the root just by clicking on links he is not supposed to and confused. Is there a notification we could use to inform that he has no access, would it be necessary?

Reply:
The notification is already there. We probably could remember from which path the user is coming from and then redirect back to it, but I think that's out of scope for this user story